sdk/digitaltwins/Azure.DigitalTwins.Core/tests/QueryTests.cs

```diff
@@ -97,13 +99,35 @@ public async Task Query_PaginationWorks()
                 // act
                 var options = new QueryOptions();
                 options.MaxItemsPerPage = pageSize;
-                AsyncPageable<string> asyncPageableResponse = client.QueryAsync(queryString, options);
+
+                var queryWaitTimeoutStopwatch = new Stopwatch();
+                bool queryHasExpectedCount = false;
+                queryWaitTimeoutStopwatch.Start();
+                while (!queryHasExpectedCount)
+                {
+                    if (queryWaitTimeoutStopwatch.ElapsedMilliseconds >= QueryWaitTimeoutMillis)
```

Comment:
Why not use a timed cancellation token source instead?

Reply:
Oh, so we keep running the query until it finally returns all the items we expect? There must be a delay in items being created and then showing up in a query. Is that right?

Reply:
Because I'm a Java developer, haha...

Reply:
I like that idea better for .NET code, though, sure

Reply:
> Oh, so we keep running the query until it finally returns all the items we expect? There must be a delay in items being created and then showing up in a query. Is that right?

That's my hunch, yes.